plugins/bitrot/transactions.go

```diff
@@ -0,0 +1,187 @@
+package bitrot
+
+import (
+	"errors"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	log "github.com/sirupsen/logrus"
+)
+
+//Returns true if there are local bricks of volume on which bitrot is enabled
+func IsBitrotAffectedNode() bool {
+	volumes, e := volume.GetVolumes()
+	if e != nil {
+		log.WithError(e).Error("Failed to get volumes")
+		return true
+	}
+	for _, v := range volumes {
+		if !v.BitrotEnabled {
+			continue
+		} else if v.State != volume.VolStarted {
+			continue
+		} else {
+			bricks := v.GetLocalBricks()
+			if len(bricks) > 0 {
+				return true
+			} else {
+				continue
+			}
+		}
+	}
+	return false
+}
+
+func ManageBitd(bitrotDaemon *Bitd) error {
+	var err error
+	AffectedNode := IsBitrotAffectedNode()
+
+	// Should the bitd and scrubber untouched on this node
+	if !AffectedNode {
+		// This condition is for disable
+		// TODO: Need to ignore errors where process is already running.
+		// Needs better handling
+		daemon.Stop(bitrotDaemon, true)
+	} else {
+		//TODO: Handle ENOENT of pidFile
+		err = daemon.Stop(bitrotDaemon, true)
+		err = daemon.Start(bitrotDaemon, true)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func ManageScrubd() error {
+	AffectedNode := IsBitrotAffectedNode()
+	scrubDaemon, err := newScrubd()
+	if err != nil {
+		return err
+	}
+
+	if !AffectedNode {
+		// This condition is for disable
+		// TODO: Need to ignore errors where process is already running.
+		// Needs better handling
+		daemon.Stop(scrubDaemon, true)
+	} else {
+		//TODO: Handle ENOENT of pidFile
+		daemon.Stop(scrubDaemon, true)
+		err = daemon.Start(scrubDaemon, true)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func txnBitrotEnable(c transaction.TxnCtx) error {
+	var volinfo volume.Volinfo
+	var startedBitd bool
+
+	// Get volInfo from ctx
+	if err := c.Get("volinfo", &volinfo); err != nil {
+		return err
+	}
+
+	// Volume is started
+	if volinfo.State != volume.VolStarted {
+		return errors.New("Volume is not started")
+	}
+
+	// bitrot already enabled
+	if volinfo.BitrotEnabled {
+		return errors.New("Bitrot already enabled on the volume")
+	}
+
+	// Set bitd status in volinfo
+	volinfo.BitrotEnabled = true
```

Comment:
Same as above. Change Volinfo only in handler before calling `Ctx.Set`

Reply:
Done